src/server/game/Entities/Object/Object.cpp

```diff
@@ -1232,11 +1236,28 @@ bool WorldObject::IsWithinLOS(float ox, float oy, float oz) const
     VMAP::IVMapManager* vMapManager = VMAP::VMapFactory::createOrGetVMapManager();
     return vMapManager->isInLineOfSight(GetMapId(), x, y, z+2.0f, ox, oy, oz+2.0f);*/
     if (IsInWorld())
-        return GetMap()->isInLineOfSight(GetPositionX(), GetPositionY(), GetPositionZ()+2.f, ox, oy, oz+2.f, GetPhaseMask());
+    {
+        Position pos;
+        if (GetTypeId() == TYPEID_PLAYER)
+            pos = { GetPositionX(), GetPositionY(), GetPositionZ() + 2.0f };
+        else
+            pos = GetHitSpherePointFor({ ox, oy, oz });
+
+        return GetMap()->isInLineOfSight(pos.GetPositionX(), pos.GetPositionY(), pos.GetPositionZ(), ox, oy, oz, GetPhaseMask());
```

Comment:
Ehm, I don't get it :x
Which position are you referring to ? ox, oy, oz or 'pos' ? I assume modified position is position get from GetHitSpherePointFor